### PR TITLE
Avoid `merge` deprecation warning on default scopes

### PR DESCRIPTION
**Pull request by ghiculescu**
Adds a failing test for https://github.com/rails/rails/pull/39328#issuecomment-860917061

I don't know how to fix it, but wanted to confirm the issue. @kamipo would love any pointers (or if you want to take it over, please do!)

### CODE REVIEW

File: activerecord/test/cases/relation/merging_test.rb [hunk 1]
```diff
@@ -157,6 +157,23 @@ def test_merge_or_clause
...
+      assert_equal [david], Author.where(id: david).merge(not_mary)
+    end
+    assert_equal [david, bob], Author.where(id: david).merge(not_mary, rewhere: true)
+
+    assert_deprecated do
+      assert_equal [], Author.where(id: mary).merge(not_mary)
+    end
+    assert_equal [david, bob], Author.where(id: mary).merge(not_mary, rewhere: true)
+  end
+
```
Comment (ghiculescu) [inline activerecord/test/cases/relation/merging_test.rb:176]: these tests all pass on `main`. they test a slightly different code path to `test_merge_not_in_clause` (`NotEqual` vs `HomogenousIn`) so I thought it was worth splitting them out.

**Comment by ghiculescu** [comment]
We've worked around this in our app (by removing the default scope) and I haven't been able to make any good progress on this issue. Closing for now. @kamipo FYI, let me know if you want me to make an issue for this.

**Comment by zzak** [comment]
@ghiculescu I wonder if we should leave this open and assign it to @kamipo. 🙏 Or if you want to open a new bug report that might help get feedback quicker.

**Comment by ghiculescu** [comment]
I created https://github.com/rails/rails/issues/42759

**Pull request closed without merging**